src/EditorFeatures/Core/Implementation/NavigationBar/NavigationBarController.cs

```diff
@@ -282,29 +258,6 @@ private void GetProjectItems(out IList<NavigationBarProjectItem> projectItems, o
                 : projectItems.First();
         }
 
-        /// <summary>
-        /// Check if the presenter has already been pushed the full model that corresponds to the
-        /// current buffer's project version stamp.
-        /// </summary>
-        private bool PresenterAlreadyHaveUpToDateFullList(CancellationToken cancellationToken)
-        {
-            AssertIsForeground();
-
-            // If it doesn't have a full list pushed, then of course not
-            if (_versionStampOfFullListPushedToPresenter == null)
-            {
-                return false;
-            }
-
-            var document = _subjectBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
-            if (document == null)
-            {
-                return false;
-            }
-
-            return document.Project.GetDependentSemanticVersionAsync(cancellationToken).WaitAndGetResult(cancellationToken) == _versionStampOfFullListPushedToPresenter;
```

Comment:
no expensive semantic checks (which can cause lots of parsing, or skeletons) happenign on UI thread.